compiler/runtime/OMRRelocationTarget.hpp

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2022 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_RELOCATION_TARGET_INCL
+#define OMR_RELOCATION_TARGET_INCL
+
+#ifndef OMR_RELOCATION_TARGET_CONNECTOR
+#define OMR_RELOCATION_TARGET_CONNECTOR
+namespace OMR { class RelocationTarget; }
+namespace OMR { typedef OMR::RelocationTarget RelocationTargetConnector; }
+#endif
+
+
+#include <stddef.h>
+#include <stdint.h>
+#include "env/TRMemory.hpp"
+#include "env/jittypes.h"
+#include "env/JitConfig.hpp"
+
+namespace TR
+{
+class RelocationRecord;
+class RelocationRuntime;
+class RelocationTarget;
+}
+
+namespace OMR
+{
+
+/**
+ * @brief The relocation target
+ * is a class that implements an API to store objects of different size.
+ * Having this class allows for abstracting away from memory implementation
+ * of an particular device and might potentially improve portability of OMR
+ * for embedded devices.
```

Comment:
This description of RelocationTarget is not accurate. It abstracts over the relocatable code and data values in JIT generated code and meta data. It has nothing to do with embedded devices, it primarily enables RelocationRecord and its subclasses to have platform independent implementations.

Reply:
Thank you, I've roughly paraphrased your explanation and inserted it to serve as a description of this class.